pkg/lib/cockpit-components-plot.jsx

```diff
@@ -0,0 +1,513 @@
+            formatter: (val) => config.formatter(val, unit_str, true)[0],
+            unit: unit_str,
+            max: ticks[ticks.length - 1]
+        };
+    } else {
+        return {
+            ticks: ticks,
+            formatter: config.formatter,
+            max: ticks[ticks.length - 1]
+        };
+    }
+}
+
+export const ZoomControls = ({ plot_state }) => {
+    function format_range(seconds) {
+        let n;
+        if (seconds >= 365 * 24 * 60 * 60) {
+            n = Math.ceil(seconds / (365 * 24 * 60 * 60));
+            return cockpit.format(cockpit.ngettext("$0 year", "$0 years", n), n);
+        } else if (seconds >= 30 * 24 * 60 * 60) {
+            n = Math.ceil(seconds / (30 * 24 * 60 * 60));
+            return cockpit.format(cockpit.ngettext("$0 month", "$0 months", n), n);
+        } else if (seconds >= 7 * 24 * 60 * 60) {
+            n = Math.ceil(seconds / (7 * 24 * 60 * 60));
+            return cockpit.format(cockpit.ngettext("$0 week", "$0 weeks", n), n);
+        } else if (seconds >= 24 * 60 * 60) {
+            n = Math.ceil(seconds / (24 * 60 * 60));
+            return cockpit.format(cockpit.ngettext("$0 day", "$0 days", n), n);
+        } else if (seconds >= 60 * 60) {
+            n = Math.ceil(seconds / (60 * 60));
+            return cockpit.format(cockpit.ngettext("$0 hour", "$0 hours", n), n);
+        } else {
+            n = Math.ceil(seconds / 60);
+            return cockpit.format(cockpit.ngettext("$0 minute", "$0 minutes", n), n);
+        }
+    }
+
+    const zoom_state = plot_state.zoom_state;
+
+    const [isOpen, setIsOpen] = useState(false);
+    useEvent(plot_state, "changed");
+    useEvent(zoom_state, "changed");
+
+    function range_item(seconds, title) {
+        return (
+            <DropdownItem key={title}
+                          onClick={() => {
+                              setIsOpen(false);
+                              zoom_state.set_range(seconds);
+                          }}>
+                {title}
+            </DropdownItem>);
+    }
+
+    if (!zoom_state)
+        return null;
+
+    return (
+        <div>
+            <Dropdown
+                isOpen={isOpen}
+                toggle={<DropdownToggle onToggle={setIsOpen}>{format_range(zoom_state.x_range)}</DropdownToggle>}
+                dropdownItems={[
+                    <DropdownItem key="now" onClick={() => { zoom_state.goto_now(); setIsOpen(false) }}>
+                        {_("Go to now")}
+                    </DropdownItem>,
+                    <DropdownSeparator key="sep" />,
+                    range_item(5 * 60, _("5 minutes")),
+                    range_item(60 * 60, _("1 hour")),
+                    range_item(6 * 60 * 60, _("6 hours")),
+                    range_item(24 * 60 * 60, _("1 day")),
+                    range_item(7 * 24 * 60 * 60, _("1 week"))
+                ]} />
+            { "\n" }
+            <Button variant="secondary" onClick={() => zoom_state.zoom_out()}
```

Comment:
Icons which don't have text but  they are wrapping icons should have an aria-label for accessibility. We can fix it in a followup.

Reply:
These icons should also have a tooltip when hovered too. Everyone should be able to understand the meaning of an icon. (Although a magnifying glass is commonly used for zooming, you'd be surprised at how many people don't understand common icons.)

Reply:
Good catch, BTW!